tests/specs/controllers/Dashboard.test.js

```diff
@@ -47,147 +56,407 @@ describe('DashboardController:', function () {
-                OFFSET,
-                PAGE_SIZE
+            const stubDisplayAppsArgs = {
+                userData: {
+                    loggedIn: true,
+                    user: {
+                        electronicEnabled: true,
+                    },
+                },
+                totalApplications: 2,
+                offset: 0,
+                sortOrder: -1,
+                currentPage: 1,
+                searchCriteria: '',
+                pageSize: 20,
+                req: reqStub,
+                res: resStub,
+                results: stubDBResults,
+            };
+
+            dashboardController._addCasebookStatusesToResults(
+                emptyCasebookResponse,
+                stubDisplayAppsArgs
             );
 
             // then
-            const expectedResult = {
-                totalPages: 2,
-                paginationMessage:
-                    'Showing 1 &ndash; 20 of 35 applications submitted in the last 60 days',
+            const expectedUserApplications = [
+                {
+                    app_status: {
+                        text: 'Not available',
+                        colorClass: 'govuk-tag--grey',
+                    },
+                    applicationtype: 'e-Apostille',
+                    doc_count: 1,
+                    main_postcode: '',
+                    payment_amount: '30.00',
+                    result_count: 2,
+                    tracking_ref: null,
+                    unique_app_id: 'A-D-21-1008-0547-D546',
+                    user_ref: 'timber',
+                },
+                {
+                    app_status: {
+                        text: 'Not available',
+                        colorClass: 'govuk-tag--grey',
+                    },
+                    applicationtype: 'e-Apostille',
+                    doc_count: 1,
+                    main_postcode: '',
+                    payment_amount: '30.00',
+                    result_count: 2,
+                    tracking_ref: null,
+                    unique_app_id: 'A-D-21-1006-2198-C15C',
+                    user_ref: 'ghfghjdf',
+                },
+            ];
+            expect(
+                resStub.view.getCall(0).args[1].users_applications
+            ).to.deep.equal(expectedUserApplications);
+        });
+    });
+
+    describe('_calculateSortParams', () => {
+        let reqStub;
+        let expectedSortParams;
+        const resStub = {};
+        const totalApplications = 2;
+        const userData = {
+            loggedIn: true,
+        };
+
+        beforeEach(() => {
+            reqStub = {
+                allParams: () => ({ dashboardFilter: null }),
+                session: {
+                    passport: {
+                        user: 123,
+                    },
+                },
+                query: {
+                    page: null,
+                    sortOrder: null,
+                    searchText: null,
+                },
```

Comment:
would a user's query string ever be `?page=null&sortOrder=null` etc? does the logic still work when `query : { }`